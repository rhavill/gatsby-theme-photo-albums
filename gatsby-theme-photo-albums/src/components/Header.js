import React from 'react'
import PropTypes from 'prop-types'
import {Link} from 'gatsby'
import {Header as ThemeHeader} from 'theme-ui'
import breadcrumbs from '../util/breadcrumbs'

const Header = ({ path, rootTitle }) => {
  const links = breadcrumbs(path, rootTitle)

  return (
    <ThemeHeader>
      <nav>{
        links.map((crumb, i) => (
          <Link key={i} to={crumb.path}>
            {crumb.title}
          </Link>))
      }
      </nav>
    </ThemeHeader>
  )
}

Header.propTypes = {
  path: PropTypes.string.isRequired,
  rootTitle: PropTypes.string,
}

Header.defaultProps = { 
  rootTitle: 'Home',
}

export default Header