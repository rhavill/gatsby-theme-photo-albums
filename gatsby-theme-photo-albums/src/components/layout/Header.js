import React from 'react'
import PropTypes from 'prop-types'
import HeaderStyles from '../../styles/Header'
import FirstBreadcrumbText from './FirstBreadcrumbText'
import Menu from './Menu'
import breadcrumbs from '../../util/breadcrumbs'

const Header = ({ path }) => {
  const links = breadcrumbs(path)
  if (links.length) {
    links[0].title = <FirstBreadcrumbText />
  }
  return (
    <HeaderStyles>
      <Menu links={links} />
    </HeaderStyles>
  )
}

Header.propTypes = {
  path: PropTypes.string.isRequired,
}

export default Header