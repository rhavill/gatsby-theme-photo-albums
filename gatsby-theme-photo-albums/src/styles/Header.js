/** @jsx jsx */
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'

const Header = ({children}) => 
  <div sx={{variant: 'layout.header'}}>
    {children}
  </div>

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default Header