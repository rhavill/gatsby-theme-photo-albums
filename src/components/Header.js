import React from 'react'
import PropTypes from 'prop-types'
import {Link} from 'gatsby'

import breadcrumbs from '../util/breadcrumbs'

const Header = ({ path, rootTitle }) => {
  const links = breadcrumbs(path, rootTitle)
  return (
    <header>
      <nav className='container'>{
        links.map((crumb, i) => (
          <Link key={i} to={crumb.path}>
            {crumb.title}
          </Link>))
      }
      </nav>
    </header>
  )
}

Header.propTypes = {
  path: PropTypes.string.isRequired,
  rootTitle: PropTypes.string,
}

Header.defaultProps = { 
  rootTitle: 'Home',
}

export default Header