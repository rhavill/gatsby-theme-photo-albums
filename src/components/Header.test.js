import React from 'react'
import renderer from 'react-test-renderer'
import {cleanup, render} from '@testing-library/react'
import Header from './Header'

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup)

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Header path='/' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('displays "title case" text for path components', () => {
    const {getByText} = render(
      <Header path='/first-dir/second-dir' />
    )
    expect(getByText('First Dir')).toBeTruthy()
    expect(getByText('Second Dir')).toBeTruthy()
  })
})
