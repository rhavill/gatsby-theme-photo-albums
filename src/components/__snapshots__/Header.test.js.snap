// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="css-1upivqb-Header"
>
  <nav>
    <a
      href="/"
    >
      Home
    </a>
  </nav>
</header>
`;
